packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -231,6 +237,11 @@ class ImageCache {
     return result;
   }
 
+  /// Returns whether this [key] has been previously added by [putIfAbsent].
+  bool containsKey(Object key) {
```

Comment:
This is breaking (we have internal customers that implement ImageCache). There are ugly ways to make it not be breaking (e.g. make `putIfAbsent` tolerate having a `loader` that returns null, which is documented as being bad but not tested against, and we have internal implementers that tolerate it).

I think it's worth the break but am willing to hear other ideas.

Reply:
Don't hesitate to make breaking changes if they are valuable. Just make sure to go through the process (which involves getting feedback about whether it's valuable).